### PR TITLE
CORE-613: Migrate META-INF/suspendable* files to OSGi agent.

### PR DESCRIPTION
**Pull request by chrisr3**
Move Quasar's built-in lists of suspendables into the instrumentation agent jar, so that they are found immediately even within an OSGi framework.

Also remove Gradle task which generates duplicate suspendables inside `quasar-core`, and fix some compiler warnings.

### CODE REVIEW

**Review by PaulHatcherR3 — APPROVED** [review]

File: quasar-core/src/main/java/co/paralleluniverse/fibers/instrument/SuspendablesScanner.java [hunk 1]
```diff
@@ -715,7 +714,7 @@ void setSupers(String superName, String[] interfaces) {
         }
 
         void setMethods(Collection<String> ms) {
-            this.methods = ms.toArray(new String[ms.size()]);
+            this.methods = ms.toArray(new String[0]);
```
Comment (mnesbit) [inline quasar-core/src/main/java/co/paralleluniverse/fibers/instrument/SuspendablesScanner.java:717]: This is confusing. Are you really sure about this?

**Review by mnesbit — COMMENTED** [review]
Can you verify my question

File: quasar-core/src/main/java/co/paralleluniverse/fibers/instrument/SuspendablesScanner.java (hunk 1 above)
Comment (chrisr3) [reply quasar-core/src/main/java/co/paralleluniverse/fibers/instrument/SuspendablesScanner.java:717]: Yes, this is the latest advice with modern JVMs.

**Review by mnesbit — APPROVED** [review]
ok

**Pull request merged**